pkgs/build-support/fetchsourcehut/default.nix

```
@@ -0,0 +1,25 @@
+{ fetchzip, lib }:
+
+{ owner
+, repo, rev
+, domain ? "sr.ht"
+, vc ? "git"
+, name ? "source"
+, ... # For hash agility
+} @ args:
+
+with lib;
+
+assert (lib.assertOneOf "vc" vc [ "hg" "git" ]);
+
+let
+  baseUrl = "https://${vc}.${domain}/${owner}/${repo}";
+
+in fetchzip (recursiveUpdate {
+  inherit name;
+  url = "${baseUrl}/archive/${rev}.tar.gz";
+  meta.homepage = "${baseUrl}/";
```

```suggestion
  meta.homepage = baseUrl;
```

On sr.ht the "/" isn't appended and e.g. `scdoc`'s man page contains "Up-to-date sources can be found at https://git.sr.ht/~sircmpwn/scdoc [...]". So we might want to change this later but it shouldn't really matter much and we could do it at any time.